### ISSUE TITLE
Failde shards in kibana

### ISSUE THREAD

**Issue by VN1977** [StamusNetworks/SELKS#255]
Hi!
We have a new installation of SELKS6. I upgraded to the last version of all components. While I open kibana I see such an error
![изображение](https://user-images.githubusercontent.com/25663601/94171061-74b9b380-fe99-11ea-93bd-2c59c652a7f8.png)
![изображение](https://user-images.githubusercontent.com/25663601/94171298-c3674d80-fe99-11ea-8122-143bd7fd099b.png)
![изображение](https://user-images.githubusercontent.com/25663601/94171334-d0843c80-fe99-11ea-826d-7e15a9b2c5f2.png)
Of course index logstash-2020..... doesn't have this field flow_id but there is scripted field Correlate Flow to form link to Evebox, so error exists.

**Comment by pevma** [comment]
Where does this error happen , on which dashboard ?

**Comment by VN1977** [comment]
It happens on standart discovery page with index pattern logstash-*

**Comment by VN1977** [comment]
Maybe this code solves the problem
```
if (doc.containsKey('flow_id'))
  { if (doc['flow_id'].size() != 0)
    {return doc['flow_id'].value}
  }
```

**Comment by VN1977** [comment]
The same trouble with FPC field.
![изображение](https://user-images.githubusercontent.com/25663601/94184796-1b5b7f80-fead-11ea-902e-04384f56c12e.png)

**Comment by VN1977** [comment]
Replaced scripted field FPC with this code
```
def src_ip = doc.containsKey('src_ip.keyword') ? (doc['src_ip.keyword'].value != null ? doc['src_ip.keyword'].value : '') : '';
def src_port = doc.containsKey('src_port') ? (doc['src_port'].value != null ? doc['src_port'].value : '') : '';
def dest_ip = doc.containsKey('dest_ip.keyword') ? (doc['dest_ip.keyword'].value != null ? doc['dest_ip.keyword'].value : '') : '';
def dest_port = doc.containsKey('dest_port') ? (doc['dest_port'].value != null ? doc['dest_port'].value : '') : '';
def proto = doc.containsKey('proto.keyword') ? (doc['proto.keyword'].value != null ? doc['proto.keyword'].value : '') : '';
return ('ip.src == ' + src_ip + ' && port.src == ' + src_port + ' && ip.dst == ' + dest_ip + ' && port.dst == ' + dest_port + ' && protocols == ' + proto.toLowerCase());
```
No errors any more

**Comment by pevma** [comment]
Thanks for the feedback!
I am thinking actually the easiest might be to just replace the whole line with `commuity_id` - https://github.com/OISF/suricata/blob/master/suricata.yaml.in#L120 and set up Moloch in a similar way the  deployments.

**Comment by VN1977** [comment]
I tried to do so.
![изображение](https://user-images.githubusercontent.com/25663601/94239292-40d2a280-ff1a-11ea-9b8d-68f0ca6543d8.png)

**Comment by pevma** [comment]
Yep - but first you need to enable it in the suricata config, restart suricata and then it will be in the newly generated log.

**Comment by VN1977** [comment]
I have enabled this option. But I had field communityid in Kibana already before. And now hothing has changed.
My mistake in scripted field was in usage of community_id instead of community_id.keyword and after correcting I don't have any errors.
But another problem is that no one Comminity ID from Kibana can be found in Moloch.

**Comment by pevma** [comment]
Think you can fix the error by importing the newest vizs form the API (or directly form the Kibana GUI ) 
https://github.com/StamusNetworks/KTS7#how-to-use

**Comment by collse** [comment]
I got a similar one on the SN-All after importing the latest

org.elasticsearch.index.fielddata.ScriptDocValues$Longs.get(ScriptDocValues.java:121)
org.elasticsearch.index.fielddata.ScriptDocValues$Longs.getValue(ScriptDocValues.java:115)
'ip == ' + doc['src_ip.keyword'].value + ' && port == ' + doc['src_port'].value + ' && ip == ' + doc['dest_ip.keyword'].value + ' && port == ' + doc['dest_port'].value + ' && protocols == ' + doc['proto.keyword'].value.toLowerCase()
                                                                         ^---- HERE

this script

'ip == ' + doc['src_ip.keyword'].value + ' && port == ' + doc['src_port'].value + ' && ip == ' + doc['dest_ip.keyword'].value + ' && port == ' + doc['dest_port'].value + ' && protocols == ' + doc['proto.keyword'].value.toLowerCase()

Lang painless
Position offset 73
Position start 0
Position end 232
Caused by type
illegal_state_exception
Caused by reason
A document doesn't have a value for a field! Use doc[<field>].size()==0 to check if a document is missing a field!

**Comment by pevma** [comment]
Can you please paste the output of 
``dpkg -l |grep stamus`` ?

**Comment by collse** [comment]
sure

dpkg -l |grep stamus 
ii  kibana-dashboards-stamus        2020042401                   amd64        Kibana 6 dashboard templates.
ii  libhtp2                         1:0.5.36-0stamus0            amd64        HTTP normalizer and parser library
ii  selks-scripts-stamus            2020121401                   amd64        SELKS Scritps by StamusN.
ii  suricata                        1:2020121102-0stamus0        amd64        Suricata open source multi-thread IDS/IPS/NSM system.

**Comment by pevma** [comment]
It seems you don’t have the latest stamus Kibana dashboards pkg. (the one you got is from April (202004...)
Can you run the “selks-upgrade_stamus” command procedure for upgrading ? (If this is prod , please try it in QA/test first )

> -- 
> Regards,
> Peter Manev

**Comment by collse** [comment]
I already had upgraded them manually and imported via kibana | https://github.com/StamusNetworks/KTS7
https://codeload.github.com/StamusNetworks/KTS7/zip/master but it seems there were key differences?

had to reset to defaults via the menu to resolve this issue - thanks

**Comment by pevma** [comment]
You can probably also use the “overwrite import “ option via the rest api too.

> -- 
> Regards,
> Peter Manev
